client/signup/steps/woocommerce-install/hooks/use-site-options/index.tsx

```diff
@@ -0,0 +1,105 @@
+import { useDispatch, useSelector } from 'react-redux';
+import {
+	requestSiteSettings,
+	saveSiteSettings,
+	updateSiteSettings,
+} from 'calypso/state/site-settings/actions';
+import { getSiteSettings } from 'calypso/state/site-settings/selectors';
+
+// WooCommerce single options.
+export const WOOP_STORE_ADDRESS_1 = 'woocommerce_store_address';
+export const WOOP_STORE_ADDRESS_2 = 'woocommerce_store_address_2';
+export const WOOP_STORE_CITY = 'woocommerce_store_city';
+export const WOOP_DEFAULT_COUNTRY = 'woocommerce_default_country';
+export const WOOP_STORE_POSTCODE = 'woocommerce_store_postcode';
+
+// Map Woop to WooCoommerce single options.
+const siteOptionsMap = {
+	address_1: WOOP_STORE_ADDRESS_1,
+	address_2: WOOP_STORE_ADDRESS_2,
+	city: WOOP_STORE_CITY,
+	postcode: WOOP_STORE_POSTCODE,
+	country: WOOP_DEFAULT_COUNTRY,
+};
+
+/**
+ * Returns the option name for the given Woop option.
+ *
+ * @param {string} option option name
+ * @returns {string} option
+ */
+function getSiteOptionName( option: string ) {
+	if ( ! ( option in siteOptionsMap ) ) {
+		return option;
+	}
+
+	return siteOptionsMap[ option ];
+}
+
+/**
+ * React custom hook to deal with single site options.
+ *
+ * @param {number} siteId - site id
+ * @returns {Array} - site option handlers
+ */
+export function useSiteOption( siteId: number ) {
+	const dispatch = useDispatch();
+
+	const settings = useSelector( ( state ) => getSiteSettings( state, siteId ) );
+
+	// Private options store.
+	const [ editedOptions, setEditedOptions ] = useState( {} );
+
+	// Dispatch the site settings request action.
+	useEffect( () => {
+		if ( ! siteId ) {
+			return;
+		}
+
+		dispatch( requestSiteSettings( siteId ) );
+	}, [ dispatch, siteId ] );
+
+	// Simple getter helper.
+	function get( option: string ) {
+		if ( ! settings || Object.keys( settings ).length === 0 ) {
+			return '';
+		}
+
+		const value = settings?.[ getSiteOptionName( option ) ] || '';
+		return value;
+	}
+
+	/*
+	 * Helper to 'update' site settings data.
+	 * The data is updated in the Redux store.
+	 */
+	const update = useCallback(
+		( option: string, value: string ) => {
+			const siteOption = getSiteOptionName( option );
+			setEditedOptions( ( state ) => ( { ...state, [ siteOption ]: value } ) );
+
+			// Store the edited option in the private store.
+			dispatch( updateSiteSettings( siteId, { [ siteOption ]: value } ) );
+		},
+		[ siteId, dispatch ]
+	);
+
+	/*
+	 * Helper to 'save' site settings data.
+	 * The data will be saved to the remote server.
+	 */
+	const save = useCallback( () => {
+		if ( ! editedOptions || ! Object.keys( editedOptions ).length ) {
+			return;
+		}
+
+		/*
+		 * Save the edited options to the server.
+		 * After the save is complete, clean the private store.
+		 */
+		dispatch( saveSiteSettings( siteId, editedOptions ) ).then( () => setEditedOptions( {} ) );
+	}, [ dispatch, editedOptions, siteId ] );
+
+	return { save, update, get };
```

Comment:
Building on the `SiteSetting` thought: Have you considered calling these functions `saveSiteSetting `, `updateSiteSetting()`, and `getSiteSetting()`? 
Looking at how they're used later on, save/update/get feel a bit ambiguous, especially once this hook gets used outside of our component.

Reply:
@allilevine @DustyReagan @Rolilink Any thoughts here?

Reply:
Looks like I'm late to this discussion 😅 More descriptive names are clearer, but `saveSiteSetting` is very close to the action name, `saveSiteSettings`. Is there an alternative?

Reply:
I don't have a strong opinion. Usually if I'm in doubt about name I lean more verbose. I think Allison has a good point about `saveSiteSetting` vs `saveSiteSettings`.

Reply:
Also late here but I lean more to `saveSiteSetting`

Reply:
something to keep in mind, for the next iterations, is the ability to deal with site options that contain only one value, for instance, a String or Number. This is the case for these new options: `woocommerce_store_address`, `woocommerce_store_address_2`, etc...

However, thinking in advance, we'll have to deal with storing data into an option that could be an Object/Array. This is the case for the onboarding profile option. It's an object, where nested fields also can be an Object/Array. The idea of adding this new hook is, also, being an abstraction layer to simplify the data handling into the components.

Then, probably, we'll have to define two hooks or check the type of the value in order to define how to store it into the site setting option. I hope it makes sense for you :intense:

Reply:
My opinion is to keep the API as it, because:

* The fact the hook returning an object instead of an array allows defining what are the methods what we want to consume clearer:
```es6
const { get, save } = useSiteSettingsOption( siteId );
```
vs
```es6
// it is not handy
const [ get, update, ...save ] = useSiteSettingsOption( siteId );
```

* The order doesn't matter
```es6
const { get, save } = useSiteSettingsOption( siteId );
```

```es6
const { save, get } = useSiteSettingsOption( siteId );
```

* It's pretty simple to rename the method with more descriptive names:
```es6
const {
  get: getSiteSetting,
  save: saveSiteSetting,
  update: updateSiteSetting,
} = useSiteSettingsOption( siteId );
```